quail/utils.py

```diff
@@ -67,10 +139,22 @@ def test_ci_output(url, vector_name, expected_file, expected_vector_name):
     robjects.r("rm(list=ls())")
 
 
-def collect_literal_inputs(request):
-    literal_inputs = [
-        request.inputs[k][0].data
-        for k in request.inputs.keys()
-        if "data_type" in vars(request.inputs[k][0]).keys()
-    ]
-    return literal_inputs
+def process_err_test(process, datainputs, err_type):
```

Comment:
I think this is a great way to factor out some of the testing code that we would see in a bunch of different places. The only issue I see here is that if our error message changes, these tests will fail. That isn't _really_ much of an issue, after all it will help us stay on top of the error messages. But we are mainly testing that the error itself is thrown, the specific message is less important.

Reply:
Okay, my only thought is that I want to ensure that all of the different exceptions are covered in each of the processes